src/core/backbone_router/local.cpp

```diff
@@ -194,7 +220,30 @@ void Local::SetState(BackboneRouterState aState)
 {
     VerifyOrExit(mState != aState, OT_NOOP);
 
+    if (mState == OT_BACKBONE_ROUTER_STATE_DISABLED)
```

Comment:
Note that previously when addresses were in `ot::Mle::Mle`,  `Mle` could only know there is BACKBONE_ROUTER_STATE changes, but it doesn't know the old state, so it would simply `RemoveUnicastAddress()` or `SubscribeMulticast()` without caring the result which might be `OT_ERROR_NOT_FOUND` or `OT_ERROR_ALREADY`.

https://github.com/openthread/openthread/blob/779da7c858445d11df571dabfe344f11670c4109/src/core/thread/mle.cpp#L1654-L1679

When moving into `BackboneRouter::Local` here, the old state is known. According to the old state `mState` and new state `aState`, it is clearly known when to add/remove/subscribe/unsubscribe. So here only do the relative update when required.

Like above explanation, we need to use the old state `mState` and new state `aState` to determine what to do exactly. I don't think sole `aState` could serve this purpose.